tests/many_to_one/tests.py

```diff
@@ -733,3 +733,13 @@ def test_reverse_foreign_key_instance_to_field_caching(self):
         child = parent.to_field_children.get()
         with self.assertNumQueries(0):
             self.assertIs(child.parent, parent)
+
+    def test_add_remove_set_by_pk(self):
+        usa = Country.objects.create(name='United States')
+        chicago = City.objects.create(name='Chicago')
+        with self.assertRaisesMessage(TypeError, "'City' instance expected, got %s" % chicago.pk):
+            usa.cities.add(chicago.pk)
+        with self.assertRaisesMessage(AttributeError, "'int' object has no attribute '_meta'"):
+            usa.cities.remove(chicago.pk)
+        with self.assertRaisesMessage(TypeError, "'City' instance expected, got %s" % chicago.pk):
```

Comment:
We can move `msg` before assertions
```python
msg = "'City' instance expected, got %s" % chicago.pk
with self.assertRaisesMessage(TypeError, msg):
     ...
with self.assertRaisesMessage(TypeError, msg):
    ...
```

Reply:
OK, let me improve `remove()` and I'll adjust.